### PR TITLE
Fixes Discovery Rule entity

### PR DESCRIPTION
**Pull request by ldjebran**
test output https://github.com/SatelliteQE/robottelo/pull/6736#issue-252762785

### CODE REVIEW

**Comment by mirzal** [comment]
I understand that after this change, test case passes. I assume it is stable, i.e. passes every time you run it.

But my concern is, code like that shouldn't be necessary. Airgun's `TextInput.value()` will call Selenium `get_attribute`, which - after going through webdriver - will eventually call JavaScript's `element[attribute]`. And for this particular field, it does work and returns expected output. At least in Chrome webdev tools:

```
> document.getElementById('search')['value']
" activation_key_id  = 4"
```

Note that JavaScript's Node object property `value` is distinct from HTML `<input>` attribute `value`. JavaScript `value` is current value of field. HTML `value` is **default** value of `<input>`. If user changes field value, JavaScript will display changed string, HTML will display old string.   
Other fields on that page are missing HTML attribute `value` as well, but they don't require this workaround.

My wild guess would be that problem is not that we need to set/read value using JavaScript directly, but that this field has autocompletion elements that somehow mess up with Airgun. Could it be that when Airgun is trying to fill next field ("host group"), it is below "search" autocompletion? Could you share some more details on when and how your test failed before this change?

--------------

On separate note, this field is AutoCompletion component from React. As of now, there are only two places in entire Satellite where this is used - one that you found here, another one on "Create Filter" page, if "Unlimited?" checkbox is not checked. But as Satellite will gradually move to React, we will probably see it in other places as well. Then, we might want to move it to `widgets.py`.

Just something for other reviewers to consider.

File: airgun/views/discoveryrule.py [hunk 1]
```diff
@@ -36,6 +36,19 @@ def is_displayed(self):
             self.title, exception=False) is not None
 
 
+class SearchInput(TextInput):
+    """Discovery Search input field, This field has no value attribute, and The only way to set or
+    read it values correctly is using javascript.
```
Comment (oshtaier) [inline airgun/views/discoveryrule.py]: we can't even read value?

Comment (ldjebran) [reply airgun/views/discoveryrule.py]: Yes the standard read, return the value from the list and not the current visible one.

**Comment by ldjebran** [comment]
Please do not merge for now some changes appeared in last snap and value attribute is rule search input field is available now and have to make some testing and perhaps some changes.

**Comment by abalakh** [comment]
I'm currently working with container tag input which has suggestions popping up with exactly the same problem @mirzal mentioned - suggestions are covering next field, causing issues. And i was planning to add extra widget for that, but it looks like there's a chance we facing the same issue. @ldjebran let's collaborate and try to create some generic widget (if our cases are actually similar).

**Comment by ldjebran** [comment]
@mirzal @oshtaier thanks for review updated the PR thanks to @abalakh for coworking on this, the problem was after fill the autocompletion list does not disappear immediately or does not disappear at all in other entities (@abalakh case).

File: airgun/widgets.py [hunk 2]
```diff
@@ -1718,3 +1718,15 @@ def fill(self, values):
                 widget = get_widget_by_name(self, key)
                 if widget:
                     widget.fill(value)
+
+
+class AutoCompleteSearchInput(TextInput):
```
Comment (abalakh) [inline airgun/widgets.py]: Any other ideas for widget name? It doesn't necessarily mean it's search input, basically any text input with auto complete, smth like 'text input with autocomplete'. `AutoCompleteTextInput` maybe? Or smth similar

File: airgun/views/discoveryrule.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
     SatTab,
 )
 from airgun.widgets import (
+    AutoCompleteSearchInput,
```
Comment (abalakh) [inline airgun/views/discoveryrule.py]: nitpick: pls sort alphabetically :)

**Comment by abalakh** [comment]
And btw, @mirzal thanks a lot for catching and bringing attention to the issue 🍻

**Comment by oshtaier** [comment]
and test result for affected discovery test case

**Comment by ldjebran** [comment]
@oshtaier changes done
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_discoveryrule.py::test_positive_end_to_end
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.26.0, timeout-1.3.3, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
collecting ... 2019-02-25 09:04:21 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                             

tests/foreman/ui_airgun/test_discoveryrule.py::test_positive_end_to_end PASSED                         [100%]

=================================== 1 passed, 2 warnings in 169.66 seconds ===================================

```

**Pull request merged**